mod-pubsub-server/src/main/java/org/folio/services/impl/SecurityManagerImpl.java

```diff
@@ -196,47 +194,58 @@ public SecurityManagerImpl(@Autowired Vertx vertx, @Autowired Cache cache,
       });
   }
 
-  private Future<Void> assignPermissions(String userId, OkapiConnectionParams params) {
-    JsonObject requestBody = new JsonObject()
-      .put("id", UUID.randomUUID().toString())
-      .put("userId", userId)
-      .put("permissions", new JsonArray(PERMISSIONS));
-    return doRequest(params, PERMISSIONS_URL, HttpMethod.POST, requestBody.encode())
-      .compose(response -> {
-        if (response.getCode() == HttpStatus.HTTP_CREATED.toInt()) {
-          LOGGER.info("Added permissions [{}] for {} user", StringUtils.join(PERMISSIONS, ","),
-            systemUserConfig.getName());
-          return Future.succeededFuture();
-        } else {
-          String errorMessage = format("Failed to add permissions %s for %s user. Received status code %s: %s",
-            StringUtils.join(PERMISSIONS, ","), systemUserConfig.getName(), response.getCode(),
-            response.getBody());
-          LOGGER.error(errorMessage);
-          return Future.failedFuture(errorMessage);
-        }
-      });
-  }
-
-  private Future<Void> addPermissions(String userId, OkapiConnectionParams params) {
-    List<Future<Void>> futures = new ArrayList<>();
-    String permUrl = PERMISSIONS_URL + "/" + userId + "/permissions?indexField=userId";
-    PERMISSIONS.forEach(permission -> {
-      JsonObject requestBody = new JsonObject()
-        .put("permissionName", permission);
-      futures.add(doRequest(params, permUrl, HttpMethod.POST, requestBody.encode())
-        .compose(response -> {
-          if (response.getCode() == HttpStatus.HTTP_OK.toInt()) {
-            LOGGER.info("Added permission {} for {} user", permission, systemUserConfig.getName());
-            return Future.succeededFuture();
-          } else {
-            String errorMessage = format("Failed to add permission %s for %s user. Received status code %s",
-              permission, systemUserConfig.getName(), response.getCode());
+  /**
+   * Create or Update permission user.
+   * @param userId user ID as known from mod-users
+   * @param params connection params.
+   * @return async result.
+   */
+  private Future<Void> establishPermissionsUser(String userId, OkapiConnectionParams params) {
+    // have to do a GET first as PUT does not allow indexField=userId.. If that was available, the
+    // GET would not be necessary.
+    return doRequest(params, PERMISSIONS_URL + "/" + userId + "?indexField=userId", HttpMethod.GET,
+      null).compose(res1 -> {
+      if (res1.getCode() == HttpStatus.HTTP_OK.toInt()) {
+        JsonObject requestBody = res1.getJson()
+          .put("permissions", new JsonArray(PERMISSIONS));
+        return doRequest(params, PERMISSIONS_URL + "/" + requestBody.getString("id"), HttpMethod.PUT,
```

Comment:
This removes any other permissions.
https://github.com/folio-org/mod-pubsub#permissions should mention this.

Reply:
Oh yes.